cmd/orb.go

```diff
@@ -740,3 +758,144 @@ https://circleci.com/orbs/registry/orb/%s
 
 	return nil
 }
+
+type OrbSchema struct {
+	Version     string                   `yaml:"version,omitempty"`
+	Description string                   `yaml:"description,omitempty"`
+	Display     interface{}              `yaml:"display,omitempty"`
+	Orbs        interface{}              `yaml:"orbs,omitempty"`
+	Commands    interface{}              `yaml:"commands,omitempty"`
+	Executors   interface{}              `yaml:"executors,omitempty"`
+	Jobs        interface{}              `yaml:"jobs,omitempty"`
+	Examples    map[string]ExampleSchema `yaml:"examples,omitempty"`
+}
+
+type ExampleUsageSchema struct {
+	Version   string      `yaml:"version,omitempty"`
+	Orbs      interface{} `yaml:"orbs,omitempty"`
+	Jobs      interface{} `yaml:"jobs,omitempty"`
+	Workflows interface{} `yaml:"workflows"`
+}
+
+type ExampleSchema struct {
+	Description string             `yaml:"description,omitempty"`
+	Usage       ExampleUsageSchema `yaml:"usage,omitempty"`
+	Result      ExampleUsageSchema `yaml:"result,omitempty"`
+}
+
+func packOrb(opts orbOptions) error {
+	// Travel our Orb and build a tree from the YAML files.
+	// Non-YAML files will be ignored here.
+	_, err := os.Stat(filepath.Join(opts.args[0], "@orb.yml"))
+	if err != nil {
+		return errors.Wrap(err, "@orb.yml file not found, are you sure this is the Orb root?")
+	}
+
+	tree, err := filetree.NewTree(opts.args[0], "executors", "jobs", "commands", "examples")
+	if err != nil {
+		return errors.Wrap(err, "An error occurred trying to build the tree")
+	}
+
+	y, err := yaml.Marshal(&tree)
```

Comment:
Yeah, I completely agree that it's a bit of a weird string of transformations. I'll see about implementing your suggestion, sounds like it could work better.

Reply:
I've done some digging/trial and error, and I don't think we can get around this without removing the patch for ordering the example stanzas in a predictable way, or repeating calls to `inlineIncludes`/`travelOrbTree`  since we can't iterate over a struct (not without some reflection anyways, which I would rather avoid). Not sure the best way to approach this outside of those two solutions, but very open to suggestions @johnswanson

Reply:
What would you think of just doing it with multiple calls to `inlineIncludes`? One each for `executors`, `commands`, and `jobs`? I think that would be cleaner - annoyingly repetitive, but closer to the intent.

Reply:
I suppose that would work, but then we remove the ability to `<<include()>>` in examples - which isn't a huge deal imo.